ocaml-lsp-server/test/ocaml_lsp_tests.ml

```diff
@@ -17,3 +17,75 @@ let%expect_test "eat_message tests" =
   [%expect {| [PASS] |}];
   test "foo bar" "foo Bar" false;
   [%expect {| [PASS] |}]
+
+let%test_module "diff" =
+  (module struct
+    let test ~from ~to_ =
+      let edits = Ocaml_lsp_server.Diff.edit ~from ~to_ in
+      let json = `List (List.map Lsp.Types.TextEdit.yojson_of_t edits) in
+      Yojson.Safe.pretty_to_string ~std:false json |> print_endline
+
+    let%expect_test "empty strings" =
+      test ~from:"" ~to_:"";
+      [%expect {| [] |}]
+
+    let%expect_test "from empty" =
+      test ~from:"" ~to_:"foobar";
+      [%expect
+        {|
+        [
+          {
+            "newText": "foobar\n",
+            "range": {
+              "end": { "character": 0, "line": 0 },
```

Comment:
Should line be 1 here perhaps?

Reply:
Hm, your comment prompted me to think more about the edits. 

I think that this particular edit looks correct because it doesn't include a newline. According to the spec: 

> If you want to specify a range that contains a line including the line ending character(s) then use an end position denoting the start of the next line.

Also, editing from an empty file to `foobar\n` in vscode, sends an edit that looks like this

```
 "contentChanges": [
        {
            "range": {
                "start": {
                    "line": 0,
                    "character": 0
                },
                "end": {
                    "line": 0,
                    "character": 0
                }
            },
            "rangeLength": 0,
            "text": "foobar\n"
        }
    ]
```

However, with this logic, test `to empty` should replace range `([0, 0], [0, 7])`, where braces denote `[line, char]`. Similar logic in `multiline`